synapse/lib/stormtypes.py

```diff
@@ -8416,7 +8416,13 @@ async def deref(self, name):
 
     async def set(self, name, valu):
         trigiden = self.valu.get('iden')
-        viewiden = self.runt.snap.view.iden
+        viewiden = self.valu.get('view')
+
+        if not viewiden:
```

Comment:
how would a trigger definition not have a `view` in it?

Reply:
I don't know but it's not a mandatory field in the schema so I thought better safe than sorry.

Reply:
Maybe it should be a mandatory field?